### PR TITLE
7127 create env

### PR DESCRIPTION
**Pull request by pankaj-cgi**
Contains code to
- Containerize the application
- Create workflows to create release and deploy in OpenShift

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ENV PBF-DATA-LOADER_HOME=/tmp
+
+#Setting the work dir as tmp coz
+WORKDIR /tmp
+
+#Copy hns-esb jar from target folder
+COPY --from=build-stage /home/app/target/*.jar /tmp/pbf-data-loader.jar
+
+
+#Start HNI-ESB
```
Comment (weskubo-cgi) [inline Dockerfile]: Should be PBF Data Loader.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+FROM adoptopenjdk:11-jre-hotspot
+
+#Setting env variable. This is used for location of external properties
+ENV PBF-DATA-LOADER_HOME=/tmp
+
+#Setting the work dir as tmp coz
+WORKDIR /tmp
+
+#Copy hns-esb jar from target folder
```
Comment (weskubo-cgi) [inline Dockerfile]: Should be pbf-data-loader.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+COPY src /home/app/src
+COPY pom.xml /home/app
+
+RUN mvn -f /home/app/pom.xml clean install
+
+
+FROM adoptopenjdk:11-jre-hotspot
+
+#Setting env variable. This is used for location of external properties
+ENV PBF-DATA-LOADER_HOME=/tmp
```
Comment (weskubo-cgi) [inline Dockerfile]: Not too important but it seems a little inconsistent to mix dashes and underscores. Underscores are preferred for env variables.

File: .github/workflows/release-pbf-data-loader.yml [hunk 4]
```diff
@@ -1,36 +1,173 @@
-# This is a basic workflow to help you get started with Actions
+# This is multi job workflow that builds the image, deploy that image in OpenShift dev environment and creates a tag in gihub repository.
```
Comment (weskubo-cgi) [inline .github/workflows/release-pbf-data-loader.yml]: Grammar/spelling:

This is a multi job workflow that builds the image, deploys that image in OpenShift dev environment and creates a tag in the github repository."

File: .github/workflows/release-pbf-data-loader.yml [hunk 5]
```diff
@@ -1,36 +1,173 @@
...
+          echo "COMMIT_SHA=${GITHUB_SHA}" | tee -a $GITHUB_ENV
+        
+      # 2. Determine tags for OpenShift image
+      - name: Determine image tags
+        run: |
+          echo "IMAGE_TAGS= ${GITHUB_SHA::12} ${{ env.RELEASE_VERSION }} ${{github.event.inputs.OPENSHIFT_IMAGE_TAG}}" | tee -a $GITHUB_ENV
+
+      # Build from docker file
+      - name: Build pbf-data-loader image from Dockerfile
+        id: build-docker-hniesb
```
Comment (weskubo-cgi) [inline .github/workflows/release-pbf-data-loader.yml]: id refers to "hniesb"

File: .github/workflows/release-pbf-data-loader.yml [hunk 6]
```diff
@@ -1,36 +1,173 @@
...
+  # This job updates the Gitops repository with image created in  this workflow. 
+  # Updating the image in Deployment.yml will result in deploying that image in OpenShift
+  Deploy:
+    runs-on: ubuntu-latest
+    needs: [createPbfDataLoaderRelease]
+    
+    steps:
+      - name: Step to move image to dev namespace
+        run: |
+          echo "This step should move the image to repository to test namespace"
```
Comment (weskubo-cgi) [inline .github/workflows/release-pbf-data-loader.yml]: Echo command refers to test namespace, not dev.

File: .github/workflows/release-pbf-data-loader.yml [hunk 7]
```diff
@@ -1,36 +1,173 @@
...
+      # 1. Login to OpenShift
+      - name: Log in to OpenShift
+        uses: redhat-actions/oc-login@v1
+        with:
+          openshift_server_url: ${{ env.OPENSHIFT_SERVER }}
+          openshift_token: ${{ env.OPENSHIFT_TOKEN }}
+          insecure_skip_tls_verify: true
+          namespace: ${{ env.DEFAULT_NAMESPACE }}
+
+      # 4. Push the image to selected repository
```
Comment (weskubo-cgi) [inline .github/workflows/release-pbf-data-loader.yml]: There's numbering here but it's inconsistent. #1 -> #4.

File: .github/workflows/release-pbf-data-loader.yml [hunk 8]
```diff
@@ -1,36 +1,173 @@
...
+        id: checkout
+        uses: actions/checkout@v3
+        with:
+          ref: 7127-create-env
 
-      # Runs a set of commands using the runners shell
-      - name: Run a multi-line script
+    # Tag and push the tag
+          # git fetch --all
+          # git pull
```
Comment (weskubo-cgi) [inline .github/workflows/release-pbf-data-loader.yml]: Not sure if these are meant to be informational or this is old commented out code.

File: .github/workflows/deploy-pbf-data-loader.yml [hunk 9]
```diff
@@ -1,36 +1,266 @@
...
+        type: choice
+        description: Select an environment to deploy
+        options:
+          - dev
+          - uat
+          - production
+      Branch:
+        description: "Enter branch name"
+        required: true
+        default: 7127-create-env
```
Comment (weskubo-cgi) [inline .github/workflows/deploy-pbf-data-loader.yml]: Should this be updated to main once testing is complete?

File: .github/workflows/deploy-pbf-data-loader.yml [hunk 10]
```diff
@@ -1,36 +1,266 @@
...
+        with:
+          script: |
+            if (`${{ github.event.inputs.Environment }}` == 'dev') {
+              return "-dev"
+            }
+            if ( `${{ github.event.inputs.Environment }}` == 'uat') {
+              return "-uat"
+            }
+            if (`${{ github.event.inputs.Environment }}` == 'production') {
+              return ""
```
Comment (weskubo-cgi) [inline .github/workflows/deploy-pbf-data-loader.yml:96]: Is this supposed to be blank for production?

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (pankaj-cgi) [reply Dockerfile]: Removed this as it is not required for pbf data loader. We are setting application properties as env variables as compared to external file in hnsesb. So this was not required.

File: .github/workflows/deploy-pbf-data-loader.yml (hunk 10 above)
Comment (pankaj-cgi) [reply .github/workflows/deploy-pbf-data-loader.yml:96]: there is no suffix for Prod resources, so this is left blank.

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**